#include "Vector.h"
#include <iostream>


/*
Исключительные ситуации генерируются:
1 - в конструкторе с параметрами, при попытке создать вектор больше максимального размера;
2 - в операции [] - при попытке обратиться к элементу вектора с индексом меньше нуля, или больше текущего размера вектора;
3 - в операции ++ - при попытке добавить элемент в заполненный вектор;
*/


int main()
{
	try
	{
		Vector b(5);
		std::cin >> b;
		std::cout << b;

		++b;
		std::cout << b;
		b++;
		std::cout << b;

		std::cout << b[10]; //генерируется исключительная ситуация
	}
	catch(const int &number)
	{
		std::cout << "Exception #" << number << " thrown!" << std::endl;
	}

	system("pause");
	return 0;
}